hoot-core/src/main/cpp/hoot/core/ops/UnconnectedWaySnapper.cpp

```diff
@@ -0,0 +1,633 @@
+  if (_snapToExistingWayNodes)
+  {
+    setMaxNodeReuseDistance(confOpts.getSnapUnconnectedWaysExistingWayNodeTolerance());
+  }
+  else
+  {
+    // Setting it to zero, if it wasn't specified, allows for skipping way node reuse completely.
+    _maxNodeReuseDistance = 0.0;
+  }
+  setMaxSnapDistance(confOpts.getSnapUnconnectedWaysSnapTolerance());
+  setAddCeToSearchDistance(confOpts.getSnapUnconnectedWaysAddCircularErrorToSearchRadius());
+  setWayDiscretizationSpacing(confOpts.getSnapUnconnectedWaysDiscretizationSpacing());
+  setMarkSnappedNodes(confOpts.getSnapUnconnectedWaysMarkSnappedNodes());
+
+  setSnapWayStatus(Status::fromString(confOpts.getSnapUnconnectedWaysSnapWayStatus()));
+  setSnapToWayStatus(Status::fromString(confOpts.getSnapUnconnectedWaysSnapToWayStatus()));
+
+  setWayToSnapCriterionClassName(confOpts.getSnapUnconnectedWaysSnapWayCriterion().trimmed());
+  setWayToSnapToCriterionClassName(confOpts.getSnapUnconnectedWaysSnapToWayCriterion().trimmed());
+  if (_snapToExistingWayNodes)
+  {
+    setWayNodeToSnapToCriterionClassName(
+      confOpts.getSnapUnconnectedWaysSnapToWayNodeCriterion().trimmed());
+  }
+
+  _conf = conf;
+}
+
+void UnconnectedWaySnapper::setMaxNodeReuseDistance(double distance)
+{
+  if (distance <= 0.0)
+  {
+    throw IllegalArgumentException(
+      "Invalid " + ConfigOptions::getSnapUnconnectedWaysExistingWayNodeToleranceKey() + " value: "
+      + QString::number(distance) + ". The value must be greater than 0.0.");
+  }
+  _maxNodeReuseDistance = distance;
+}
+void UnconnectedWaySnapper::setMaxSnapDistance(double distance)
+{
+  if (distance <= 0.0)
+  {
+    throw IllegalArgumentException(
+      "Invalid " + ConfigOptions::getSnapUnconnectedWaysSnapToleranceKey() + " value: "
+      + QString::number(distance) + ". The value must be greater than 0.0.");
+  }
+  _maxSnapDistance = distance;
+}
+void UnconnectedWaySnapper::setWayDiscretizationSpacing(double spacing)
+{
+  if (spacing <= 0.0)
+  {
+    throw IllegalArgumentException(
+      "Invalid " + ConfigOptions::getSnapUnconnectedWaysDiscretizationSpacingKey() + " value: "
+      + QString::number(spacing) + ". The value must be greater than 0.0.");
+  }
+  _snapToWayDiscretizationSpacing = spacing;
+}
+
+void UnconnectedWaySnapper::setWayToSnapToCriterionClassName(QString name)
+{
+  if (name.trimmed().isEmpty())
+  {
+    LOG_WARN(
+      "No snap to way criterion specified for the Unconnected Way Snapper.  " <<
+      "Defaulting to: hoot::WayCriterion.");
+    _wayToSnapToCriterionClassName = "hoot::WayCriterion";
+  }
+  else
+  {
+    _wayToSnapToCriterionClassName = name.trimmed();
+  }
+}
+void UnconnectedWaySnapper::setWayToSnapCriterionClassName(QString name)
+{
+  if (name.trimmed().isEmpty())
+  {
+    LOG_WARN(
+      "No snap way criterion specified for the Unconnected Way Snapper.  " <<
+      "Defaulting to: hoot::WayCriterion.");
+    _wayToSnapCriterionClassName = "hoot::WayCriterion";
```

Comment:
This default ought to come from `ConfigOptions::getSnapUnconnectedWaysSnapWayCriterionDefaultValue()` instead of hard coding the string `hoot::WayCriterion`, in the warning and the actual assignment of the class name.